packages/block-library/src/image/edit.native.js

```diff
@@ -76,6 +79,89 @@ const getUrlForSlug = ( image, sizeSlug ) => {
 	return image?.media_details?.sizes?.[ sizeSlug ]?.source_url;
 };
 
+function LinkSettings( {
+	attributes,
+	dismissSheet,
+	image,
+	isLinkSheetVisible,
+	setMappedAttributes,
+} ) {
+	const route = useRoute();
+	const { href: url, ...unMappedAttributes } = attributes;
+	const mappedAttributes = { ...unMappedAttributes, url };
+
+	// Persist attributes passed from child screen
+	useEffect( () => {
+		const { linkDestination, url: newUrl } = route.params || {};
+		if ( linkDestination || newUrl ) {
+			setMappedAttributes( {
+				url: newUrl,
+				linkDestination,
+			} );
+		}
+	}, [ route.params?.url, route.params?.linkDestination ] );
+
+	let valueMask;
+	switch ( mappedAttributes.linkDestination ) {
+		case LINK_DESTINATION_MEDIA:
+			valueMask = __( 'Media File' );
+			break;
+		case LINK_DESTINATION_ATTACHMENT:
+			valueMask = __( 'Attachment Page' );
+			break;
+		case LINK_DESTINATION_CUSTOM:
+			valueMask = __( 'Custom URL' );
+			break;
+		default:
+			valueMask = __( 'None' );
+			break;
+	}
+
+	const linkSettingsOptions = {
+		url: {
+			valueMask,
+			autoFocus: false,
+			autoFill: true,
+		},
+		openInNewTab: {
+			label: __( 'Open in new tab' ),
+		},
+		linkRel: {
+			label: __( 'Link Rel' ),
+			placeholder: _x( 'None', 'Link rel attribute value placeholder' ),
+		},
+	};
+
+	return (
+		<PanelBody title={ __( 'Link Settings' ) }>
+			<LinkSettingsNavigation
+				isVisible={ isLinkSheetVisible }
+				url={ mappedAttributes.url }
+				rel={ mappedAttributes.rel }
+				label={ mappedAttributes.label }
+				linkTarget={ mappedAttributes.linkTarget }
```

Comment:
Some history for context about `mappedAttributes`: originally, the interface to link settings was more narrow, and `attributes` were passed as a "bag of properties" to the component. Most usages included an attribute named `url` for the link url (e.g. format library, button block, etc.), but image block uses the attribute `href` for this. `mappedAttributes` and `setMappedAttributes` was a way to adapt the usage for image block to comply with the API.

This was changed [in a subsequent PR](https://github.com/WordPress/gutenberg/commit/db6d666f3165e78a4dcf350c7ff7f6638e8439e0#diff-7126ae03000a022bcefa0d2a5fb03b6aca56ed4047e3a49d2131babf229cf6c1R380-R383), though, for performance reasons, and the API surface area was expanded to include the attributes individually as props. An opportunity was missed her, imo, to remove the `mappedAttributes` abstraction, as it is no longer necessary.

Since we are refactoring this to an extracted component here, this is a good chance to simplify this, imo, and get rid of `mappedAttributes` (i.e. we can just directly destructure the properties we need from `attributes` [here](https://github.com/WordPress/gutenberg/pull/34846/files#diff-7126ae03000a022bcefa0d2a5fb03b6aca56ed4047e3a49d2131babf229cf6c1R90-R91)). Wdyt?

Reply:
Thanks for the suggestion. I addressed this in https://github.com/WordPress/gutenberg/pull/34846/commits/2d7712167ccba131e1bba4f2957d473569516c5d.